src/stdio/rename.c

```diff
@@ -6,7 +6,11 @@ int rename(const char *old, const char *new)
 {
 #ifdef SYS_rename
 	return syscall(SYS_rename, old, new);
+#else
+#ifdef SYS_renameat2
```

Comment:
Possible typo: This one uses the new syscall if both old and new exist, but the mmap/renameat changes prefer the old one.

(I'm not a musl or riscv maintainer, I'm just a random nobody who found this PR through your blog <https://drewdevault.com/2018/12/20/Porting-Alpine-Linux-to-RISC-V.html>. Hi.)

Reply:
Good eye. I took the more conservative approach with this one based on the ifdefs which were already here - they prefer rename over renameat even though the latter is newer/more flexible.

Reply:
worth leaving a comment in the code here to that effect? Anytime something is “surprising”  (as the commenter was) usually qualifies IMO.

Reply:
I'll mention this when it comes up in code review - but since I didn't write the original code, I don't think it's appropriate for me to speculate on the intentions of the author.

Reply:
Indeed, SYS_renameat needs to be used rather than SYS_renameat2 if both are defined on the arch, since there exist old kernels were SYS_renameat2 is not present (ENOSYS). These changes would make musl stop working on those kernels. Also, from a style perspective, `#elif defined(...` is probably preferable to avoid gratuitously nested `#ifdef` (harder to read). These changes are not riscv-specific but necessary for support of any new arch where the old syscall isn't present, and should be merged upstream as prerequisites for the riscv port.